client/ayon_core/tools/push_to_project/control.py

```diff
@@ -207,22 +217,34 @@ def wait_for_process_thread(self):
         self._process_thread = None
 
     def _prepare_source_label(self):
-        if not self._src_project_name or not self._src_version_id:
+        if not self._src_project_name or not self._src_version_ids:
             return "Source is not defined"
 
         folder_entity = self._src_folder_entity
         if not folder_entity:
             return "Source is invalid"
 
+        no_of_products = len(self._src_product_entities)
+        no_of_versions = len(self._src_version_entities)
+        if no_of_products != no_of_versions:
```

Comment:
This doesn't make any sense? Why would you require to have the same number of versions and products? And how do you know the order does match? You should find product entity based on `"productId"` stored on version entity.